content/docs/user-guide/data-pipelines/defining-pipelines.md

```diff
@@ -0,0 +1,253 @@
+stages of the <abbr>pipeline</abbr> are also reproduced.
+
+<admon type="info">
+
+DVC [calculates a hash] of file/dir contents to compare vs. previous versions.
+This is a distinctive mechanism over traditional build tools like `make`.
+
+[calculates a hash]:
+  /doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory
+
+</admon>
+
+File system-level dependencies are defined in the `deps` field of `dvc.yaml`
+stages; Alternatively, using the `--deps` (`-d`) option of `dvc stage add` (see
+the previous section's example).
+
+<details>
+
+### External dependencies: click to learn more.
+
+A less common kind of dependency is a _URL dependency_. Instead of files in a
+local disk, you can `dvc import` data from another <abbr>DVC project</abbr> (for
+example hosted on GitHub). External dependencies establish relationships between
+different projects or systems (see `dvc import-url`).
+[Get all the details](/doc/user-guide/external-dependencies).
+
+<admon type="info">
+
+DVC will use special methods to check whether the contents of an URL have
+changed for the purpose of stage invalidation.
+
+</admon>
+
+</details>
+
+## Parameter dependencies
+
+A more granular type of dependency is the parameter (`params` field of
+`dvc.yaml`), or _hyperparameters_ in machine learning. These represent simple
+values used inside your code to tune data processing, or that affect stage
+execution in any other way. For example, training a [Neural Network] usually
+requires _batch size_ and _epoch_ values.
+
+Instead of hard-coding param values, your code can read them from a structured
+file (e.g. YAML format). DVC can track any key/value pair in a supported
+[parameters file] (`params.yaml` by default). Params are granular dependencies
+because DVC only invalidates stages when the corresponding part of the params
+file has changed.
+
+```yaml
+stages:
+  train:
+    cmd: ...
+    deps: ...
+    params: # from params.yaml
+      - learning_rate
+      - nn.epochs
+      - nn.batch_size
+    outs: ...
+```
+
+<admon type="info">
+
+See [more details] about this syntax.
+
+</admon>
+
+Use `dvc params diff` to compare parameters across project versions.
+
+[parameters file]:
+  /doc/user-guide/project-structure/dvcyaml-files#parameters-files
+[neural network]:
+  https://machinelearningmastery.com/difference-between-a-batch-and-an-epoch/
+[more details]: /doc/user-guide/project-structure/dvcyaml-files#parameters
+
+## Outputs
+
+Stage outputs are files written by <abbr>pipelines</abbr> such as machine
+learning models, intermediate artifacts, as well as data [plots] and performance
+[metrics]. These files are <abbr>cached</abbr> by DVC automatically, and tracked
+with the help of `dvc.lock` files (or `.dvc` files, see `dvc add`).
+
+Output definitions can connect stages to one another (as explained earlier). So
+when they change, DVC may need to reproduce subsequent stages as well (handled
+automatically).
```

Comment:
Maybe be more specific here, like `Outputs can be dependencies of other stages`.

Reply:
Updated.